newrelic/newrelic.go

```diff
@@ -85,10 +86,14 @@ func ConfigAdminAPIKey(adminAPIKey string) ConfigOption {
 }
 
 // ConfigRegion sets the New Relic Region this client will use.
-func ConfigRegion(region config.RegionType) ConfigOption {
+func ConfigRegion(r region.Name) ConfigOption {
 	return func(cfg *config.Config) error {
-		cfg.Region = region
-		return nil
+		if region, ok := region.Regions[r]; ok {
+			regCopy := *region
```

Comment:
What's the need for the copy?  Just to safeguard against runtime changes to region?

Reply:
Exactly,  Since we can not make Struct constants, always copy because user can change the result and have side effects otherwise.